### ISSUE TITLE
TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by omarkhatibco** [notiz-dev/prisma-dbml-generator#14]
Hi, 

I was trying to test it against my prisma schema und I got this error.

```bash 
TypeError: Cannot read property 'name' of undefined
    at joinField (node_modules/prisma-dbml-generator/dist/generator/many-to-many-tables.js:27:23)
    at /node_modules/prisma-dbml-generator/dist/generator/many-to-many-tables.js:24:33
```

I can share the schema, if it's important.

**Comment by marcjulian** [comment]
@omarkhatibco yes your schema or the models which creates this error would be helpful to investigate. Seems like it's an error about a many-to-many relationship.

**Comment by omarkhatibco** [comment]
Hi @marcjulian 


```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["nApi", "selectRelationCount", "filterJson"]
}

generator nestgraphql {
  provider = "node node_modules/prisma-nestjs-graphql"
  output   = "../src/@generated/"
  reExport = All
}

generator dbml {
  provider = "prisma-dbml-generator"
}

generator docs {
  provider = "node node_modules/prisma-docs-generator"
}

enum UserStatus {
  ACTIVE
  FUTURUSER
  INACTIVE
  INVITED
}

enum FriendshipStatus {
  REQUESTED
  ACCEPTED
  REJECTED
}

model User {
  id String @id @default(cuid())

  // Auth Fields
  mobileNumber String  @unique
  email        String  @unique
  /// @HideField({ output: true, input: true })
  pin          String?

  // Personal Information
  firstName String?
  lastName  String?
  birthday  DateTime?


  // System Data
  status    UserStatus @default(INACTIVE)
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt

  //Associated Profile
  profile Profile?

  //Associated Image
  profileImage UserFile?


  //Friends - relations n-m
  friends  User[] @relation("UserFriendsUser", references: [id])
  friendOf User[] @relation("UserFriendsUser", references: [id])

  //FriendsRequest
  sentFriendsRequest     FriendRequest[] @relation("UserSendsFriendRequest")
  receivesFriendsRequest FriendRequest[] @relation("UserReceivesFriendRequest")

  //parent and child for FUTURE USER
  parent   User?   @relation("UserParentOfUser", fields: [parentId], references: [id])
  children User[]  @relation("UserParentOfUser")
  parentId String?

  // Associated Posts - relations 1-n
  posts         Post[] @relation("UserAuthorsPost")
  receivedPosts Post[] @relation("userReceivesPosts")

  // Comments
  comments Comment[]

  // created Moods
  moods Mood[]

}

model FriendRequest {
  user     User             @relation("UserSendsFriendRequest", fields: [userId], references: [id])
  userId   String
  friend   User             @relation("UserReceivesFriendRequest", fields: [userId], references: [id])
  friendId String
  status   FriendshipStatus

  @@id([userId, friendId])
}

model Profile {
  id String @id @default(cuid())

  // notification Settings
  notificationStatus    Boolean?  @default(true)
  notificationStartTime DateTime?
  notificationEndTme    DateTime?

  //Extras
  tos       DateTime @default(now())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  // Relations
  user   User   @relation(fields: [userId], references: [id])
  userId String @unique

  currentMode   Mood?   @relation(fields: [currentModeId], references: [id])
  currentModeId String?

}

model Post {
  id String @id @default(cuid())

  // Type - Content
  content  String?
  location Json?
  youtube  String[]
  spotify  String[]

  // Type - Media
  media PostFile[]


  // Extras
  publishedAt DateTime
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt

  // relations with User
  author     User   @relation("UserAuthorsPost", references: [id], fields: [authorId])
  authorId   String
  receivedBy User[] @relation("userReceivesPosts", references: [id])

  // Publish Options
  receivedAt     DateTime?
  receivedMood   Mood?     @relation(fields: [receivedMoodId], references: [id])
  receivedMoodId String?

  // comments
  comments Comment[]

}

model Comment {
  id      String @id @default(cuid())
  // Type - Content
  content String

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt


  //ParentPost
  post   Post   @relation(fields: [postId], references: [id])
  postId String

  // Comments
  parentComment    Comment?  @relation("CommentToComment", fields: [parentCommentId], references: [id])
  childrenComments Comment[] @relation("CommentToComment")
  parentCommentId  String?

  // Author
  author   User   @relation(references: [id], fields: [authorId])
  authorId String
}

model Mood {
  id String @id @default(cuid())

  // Data
  name String

  //Extras
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  author   User?   @relation(fields: [authorId], references: [id])
  authorId String?

  // connection to mood
  profiles Profile[]
  posts    Post[]
}

model PostFile {
  id  String @id
  url String

  post   Post   @relation(fields: [postId], references: [id])
  postId String
}

model UserFile {
  id  String @id
  url String

  user   User   @relation(fields: [userId], references: [id])
  userId String
}

model Notification {
  id String @id @default(cuid())

}

```

**Comment by marcjulian** [comment]
The many-to-many relation between `userReceivesPosts` is breaking the generation. If the relation is specified using `receivedBy User[] @relation("userReceivesPosts", references: [id])receivedBy User[] @relation("userReceivesPosts", references: [id])` the second part of the relationship can not be found in th DMMF (Data Model Meta Format) provided by Prisma for the generator. I will add a check that the error won't be thrown and the DMBL is still created. However, it won't contain the many-to-many table for `userReceivesPosts`. 

Thanks for reporting this, I will add a test for it.

Here are two solution for your schema: If you remove th `@relation` then it is a [implicit m-n-relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#implicit-many-to-many-relations). If you want to specifiy the name of the relationship then us `@relation("userReceivesPosts")` on both relations.

```diff
- receivedBy User[] @relation("userReceivesPosts", references: [id])
+ receivedBy User[] @relation("userReceivesPosts")
```

**Comment by marcjulian** [comment]
Closed with d5d3494, please try out the latest dev release

```bash
npm install -D prisma-dbml-generator@dev
```

**Comment by omarkhatibco** [comment]
@marcjulian, thanks for your help !
